src/vm.rs

```diff
@@ -65,6 +70,31 @@ fn detect_cpu_freq() -> u32 {
 	mhz
 }
 
+/// Generates a random guest address for Uhyve's virtualized memory, provided that the feature is enabled.
+/// For this purpose, ThreadRng is used. Currently, this feature only works on Linux (x86_64).
+/// 
+/// This function gets invoked when a new UhyveVM gets created, provided that the object file is relocatable.
+fn generate_address(object_mem_size: usize, params_mem_size: usize) -> u64 {
+	#[cfg(feature = "aslr")]
+	#[cfg(not(all(target_arch = "x86_64", target_os = "linux")))]
+	compile_error!("ASLR is only supported on Linux (x86_64)");
+
+	#[cfg(feature = "aslr")]
+	{
+		// TODO: Investigate soundness.
+		// TODO: Implement more secure alternatives.
+		let mut rng = rand::thread_rng();
+		let start_address_upper_bound: u64 = 0x0000_FFFF_FFFF_0000 - object_mem_size as u64 - KERNEL_OFFSET as u64 - params_mem_size as u64;
```

Comment:
This constant was changed from 0x000F_FFFF_FFFF_0000 to 0x0000_FFFF_FFFF_0000 for testing purposes and should be reverted.